### PR TITLE
Introduce new JSON type to property

### PR DESCRIPTION
**Pull request by GDLMadushanka**
properties
Fixes wso2/product-ei/issues/5266

### CODE REVIEW

File: modules/core/src/main/java/org/apache/synapse/util/xpath/SynapseJsonPath.java [hunk 1]
```diff
@@ -91,7 +98,22 @@ public SynapseJsonPath(String jsonPathExpression)  throws JaxenException {
         EIPUtils.setJsonPathConfiguration();
 
         this.contentAware = true;
-        this.expression = jsonPathExpression;
+        // supporting evaluation against a property
+        // Ex: json-eval($ctx:prop1.student.name)
+        Pattern fullPattern = Pattern.compile(JSON_PATH_WITH_PROP_REGEX);
```
Comment (tishan89) [inline modules/core/src/main/java/org/apache/synapse/util/xpath/SynapseJsonPath.java:103]: Shall we to a small test to compare this with Startswith()

File: modules/core/src/main/java/org/apache/synapse/util/xpath/SynapseJsonPath.java [hunk 2]
```diff
@@ -125,6 +147,20 @@ public String stringValueOf(MessageContext synCtx) {
...
         }
+
+        // evaluating the jsonPath against a property
+        if (propertyExpression != null) {
+            try {
+                SynapseXPath xPath = new SynapseXPath(propertyExpression);
+                String result = xPath.stringValueOf(synCtx);
+                return stringValueOf(result);
+            } catch (JaxenException e) {
+                log.error("Xpath evaluation failed", e);
```
Comment (tishan89) [inline modules/core/src/main/java/org/apache/synapse/util/xpath/SynapseJsonPath.java:158]: log the expression too

**Review by tishan89 — CHANGES_REQUESTED** [review]

File: modules/core/src/main/java/org/apache/synapse/util/xpath/SynapseJsonPath.java (hunk 2 above)
Comment (GDLMadushanka) [reply modules/core/src/main/java/org/apache/synapse/util/xpath/SynapseJsonPath.java]: +1 fixed

File: modules/core/src/main/java/org/apache/synapse/util/xpath/SynapseJsonPath.java (hunk 1 above)
Comment (GDLMadushanka) [reply modules/core/src/main/java/org/apache/synapse/util/xpath/SynapseJsonPath.java]: +1 fixed

**Review by tishan89 — APPROVED** [review]

**Pull request merged**